sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricsTest.java

```diff
@@ -121,62 +123,89 @@ public void counterToCell() {
     assertThat(cell.getCumulative(), CoreMatchers.equalTo(42L));
   }
 
-  @Category({ValidatesRunner.class, UsesCommittedMetrics.class})
+  @Category({ValidatesRunner.class, UsesCommittedMetrics.class, UsesCounterMetrics.class,
+      UsesDistributionMetrics.class, UsesGaugeMetrics.class})
   @Test
-  public void committedMetricsReportToQuery() {
+  public void testAllCommittedMetrics() {
     PipelineResult result = runPipelineWithMetrics();
-
     MetricQueryResults metrics = result.metrics().queryMetrics(MetricsFilter.builder()
         .addNameFilter(MetricNameFilter.inNamespace(MetricsTest.class))
         .build());
 
-    assertThat(metrics.counters(), hasItem(
-        committedMetricsResult(NAMESPACE, "count", "MyStep1", 3L)));
-    assertThat(metrics.distributions(), hasItem(
-        committedMetricsResult(NAMESPACE, "input", "MyStep1",
-            DistributionResult.create(26L, 3L, 5L, 13L))));
+    testAllMetrics(metrics, true);
+  }
 
-    assertThat(metrics.counters(), hasItem(
-        committedMetricsResult(NAMESPACE, "count", "MyStep2", 6L)));
-    assertThat(metrics.distributions(), hasItem(
-        committedMetricsResult(NAMESPACE, "input", "MyStep2",
-            DistributionResult.create(52L, 6L, 5L, 13L))));
-    assertThat(metrics.gauges(), hasItem(
-        committedMetricsResult(NAMESPACE, "my-gauge", "MyStep2",
-            GaugeResult.create(12L, Instant.now()))));
+  @Category({ValidatesRunner.class, UsesAttemptedMetrics.class, UsesCounterMetrics.class,
+      UsesDistributionMetrics.class, UsesGaugeMetrics.class})
+  @Test
+  public void testAllAttemptedMetrics() {
+    PipelineResult result = runPipelineWithMetrics();
+    MetricQueryResults metrics = result.metrics().queryMetrics(MetricsFilter.builder()
```

Comment:
This piece of code seems appears 8 times in `MetricsTest`:
```java
MetricQueryResults metrics = result.metrics().queryMetrics(MetricsFilter.builder()
    .addNameFilter(MetricNameFilter.inNamespace(MetricsTest.class))
    .build());
```
Perhaps it is worth a helper method such as 
```java
MetricQueryResults queryMetrics(PipelineResult result) { ... }
```